step03/models.py

```diff
@@ -0,0 +1,49 @@
+from sqlalchemy import Column, String, Integer, ForeignKey, Boolean
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import relationship
+
+Base = declarative_base()
+
+
+class Profiles(Base):
+  __tablename__ = 'user_profiles'
+  user_number = Column(Integer, primary_key=True)
```

Comment:
I would make username (or email) the primary key and do away with user_number because username (or email) should be uniquely identifying.

Additionally, I would make github_id another primary_key so you can access it directly using [query.get()](https://docs.sqlalchemy.org/en/13/orm/query.html#sqlalchemy.orm.query.Query).

Alternatively, you can create another table matching user_id to github_id (for those who registered using  github) and use ForeignKey to create one-to-one mappings [docs](https://docs.sqlalchemy.org/en/13/orm/basic_relationships.html#one-to-one). The upside of this is that when you add another auth source (e.g. google) you don't have to change the profile model to have another google_id primary key, meaning migrations. The downside is that you now have two tables.